src/stac_task/models.py

```diff
@@ -0,0 +1,219 @@
+"""Basic models, with little to no logic.
+
+If models are more complicated, they should go in their own module (aka file).
+"""
+
+from __future__ import annotations
+
+from typing import Any, Dict, List, Literal, Optional, Union
+
+import pystac
+import pystac.utils
+from pydantic import BaseModel, ConfigDict
+
+
+class Anything(BaseModel):
+    """A model for any dictionary."""
+
+    model_config = ConfigDict(extra="allow")
+
+
+class UploadOptions(BaseModel):
+    """Options for uploading items and assets after payload execution."""
+
+    path_template: str = r"${collection}"
+    """A string template for specifying the location of uploaded assets.
+    
+    See
+    https://pystac.readthedocs.io/en/stable/api/layout.html#pystac.layout.LayoutTemplate
+    for the available fields.
+    """
+
+    public_assets: Optional[Union[Literal["ALL"], List[str]]] = None
```

Comment:
we removed the use of the public_assets, which is really specific to AWS.  
The upload options here are basically the options that should be passed through to stac-asset.

Reply:
Okay, I was doing my best to maintain backwards compatibility with the documented options. I'll remove this one though, or should we warn folks that we're ignoring it, if the field is present?

Reply:
AFAIK no one is using it (none of our projects are).  In the cases where we are publishing public assets we've switched to setting the bucket policy, rather than using a per-Item ACL.
I think this PR is a big enough change that we don't need to worry about maintaining backward compatibility, but we should include a list of update notes of things that would need to change moving from old stac-task to the new version.